File: android/core/data/src/main/java/com/app2/core/data/remote/dto/NotificationDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class NotificationDTO(
    val id: Int,
    val type: String,
    val title: String? = null,
    val message: String? = null,
    @SerialName("product_id") val productId: Int? = null,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    @SerialName("warehouse_name") val warehouseName: String? = null,
    @SerialName("is_read") val isRead: Int? = null,
    @SerialName("link_type") val linkType: String? = null,
    @SerialName("link_id") val linkId: Int? = null,
    @SerialName("created_at") val createdAt: String? = null
)

File: ios/StockPro/Data/DTOs/NotificationDTOs.swift
import Foundation

struct NotificationDTO: Decodable, Identifiable {
    let id: Int
    let message: String
    let type: String?
    let is_read: Bool
    let warehouse_id: Int?
    let warehouse_name: String?
    let link_type: String?
    let link_id: Int?
    let created_at: String
}
